### PR TITLE
Kill fedmsg

### PR DESCRIPTION
**Pull request by abompard**
This changeset removes Fedmsg.

The first commit only removes the live updater from the UI, the second removes all of fedmsg from Bodhi.

### CODE REVIEW

File: bodhi/server/migrations/env.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (bowlofeggs) [inline bodhi/server/migrations/env.py]: We shouldn't need this with Python 3, right?

File: bodhi/server/notifications.py [hunk 2]
```diff
@@ -15,201 +15,89 @@
...
     """
     Send messages via AMQP after a database commit occurs.
 
     Args:
         session (sqlalchemy.orm.session.Session): The session that was committed.
     """
-    if 'messages' in session.info:
-        for m in session.info['messages']:
+    if "messages" in session.info:
+        for m in session.info["messages"]:
```
Comment (bowlofeggs) [inline bodhi/server/notifications.py]: This seems like an unrelated style change.

File: bodhi/server/notifications.py [hunk 3]
```diff
@@ -15,201 +15,89 @@
...
-                _log.debug('Emitted a fedmsg, %r, on the "%s" topic, queued by %r',
-                           msg, topic, session)
-            # Tidy up after ourselves so a second call to commit on this session won't
-            # send the same messages again. We cannot delete topic from fedmsg dict
-            # because we cannot change dictionary size during iteration
-            session.info['fedmsg'][topic] = []
-
-
-@event.listens_for(Session, 'after_commit')
+@event.listens_for(Session, "after_commit")
```
Comment (bowlofeggs) [inline bodhi/server/notifications.py]: This seems like an unrelated style change.

File: bodhi/server/notifications.py [hunk 4]
```diff
@@ -15,201 +15,89 @@
...
             try:
                 api.publish(m)
             except fml_exceptions.BaseException:
                 # In the future we should handle errors more gracefully
-                _log.exception("An error occurred publishing %r after a database commit", m)
-        session.info['messages'] = []
+                _log.exception(
+                    "An error occurred publishing %r after a database commit", m
+                )
+        session.info["messages"] = []
```
Comment (bowlofeggs) [inline bodhi/server/notifications.py]: This seems like an unrelated style change.

File: bodhi/server/notifications.py [hunk 5]
```diff
@@ -15,201 +15,89 @@
...
             currently active database transaction successfully commits. If true,
             the messages is sent immediately.
     """
-    if bodhi.server.config.config.get('fedmsg_enabled'):
-        _fedmsg_publish(topic, msg, force)
-        return
+    # Dirty, nasty hack that I feel shame for: use the fedmsg encoder that modifies
+    # messages quietly if they have objects with __json__ methods on them.
+    # For now, copy that behavior. In the future, callers should pass
+    # fedora_messaging.api.Message sub-classes or this whole API should go away.
```
Comment (bowlofeggs) [inline bodhi/server/notifications.py:66]: Now that we actually have Message classes, I do plan to do this soon!

File: bodhi/server/notifications.py [hunk 6]
```diff
@@ -15,201 +15,89 @@
...
     session = Session()
-    if 'messages' not in session.info:
-        session.info['messages'] = []
-    session.info['messages'].append(message)
-    _log.debug('Queuing message %r for delivery on session commit', message.id)
-
+    if "messages" not in session.info:
+        session.info["messages"] = []
+    session.info["messages"].append(message)
+    _log.debug("Queuing message %r for delivery on session commit", message.id)
```
Comment (bowlofeggs) [inline bodhi/server/notifications.py]: This seems like an unrelated style change.

File: bodhi/tests/server/test_notifications.py [hunk 7]
```diff
@@ -201,99 +79,79 @@ def test_no_messages(self):
     def test_clear_messages_on_send(self):
         """Assert the message queue is cleared after the event handler runs."""
         session = Session()
-        session.info['messages'] = [api.Message()]
+        session.info["messages"] = [api.Message()]
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_notifications.py]: This seems like an unrelated style change.

File: bodhi/tests/server/test_notifications.py [hunk 8]
```diff
@@ -201,99 +79,79 @@ def test_no_messages(self):
...
 
         with fml_testing.mock_sends(api.Message()):
             notifications.send_messages_after_commit(session)
 
-        self.assertEqual(session.info['messages'], [])
+        self.assertEqual(session.info["messages"], [])
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_notifications.py]: This seems like an unrelated style change.

I'll stop pointing these out, but let's make the commits atomic and avoid style changes.

File: bodhi/tests/server/test_notifications.py [hunk 9]
```diff
@@ -201,99 +79,79 @@ def test_no_messages(self):
...
+        )
+
+    def test_default_time_struct(self):
+        """Time structures should be converted to timestamps."""
+        timestamp = 1553016625
+        dt = datetime.datetime.fromtimestamp(timestamp)
+        self.assertEqual(
+            timestamp,
+            notifications.FedMsgEncoder().default(dt.timetuple())
+        )
```
Comment (bowlofeggs) [inline bodhi/tests/server/test_notifications.py]: Do we have any code that sends these time and set objects over messages?

File: docs/developer/releases.rst [hunk 10]
```diff
@@ -219,31 +218,31 @@ To build the beta, follow these steps:
...
+
      # The language for content autogenerated by Sphinx. Refer to documentation
      # for a list of supported languages.
     diff --git a/setup.py b/setup.py
     index 44566ff5..74297bb5 100644
     --- a/setup.py
     +++ b/setup.py
     @@ -42,7 +42,7 @@ def get_requirements(requirements_file='requirements.txt'):
-     
+
```
Comment (bowlofeggs) [inline docs/developer/releases.rst]: There are a lot of unrelated style changes in this file as well. Actually I do like removing unneeded whitespace, but let's do it in a separate pull request.

File: docs/server_api/index.rst [hunk 11]
```diff
@@ -2,12 +2,8 @@
...
 =================
 
-fedmsg API
-==========
-
-The fedmsgs bodhi sends
-`are documented by the Fedora fedmsg project <https://fedora-fedmsg.readthedocs.io/en/latest/topics.html#bodhi>`_.
-
+.. TODO: add Fedora Messaging documentation when we can generate it from the schemas.
+   See https://github.com/fedora-infra/fedora-messaging/issues/120
```
Comment (bowlofeggs) [inline docs/server_api/index.rst]: Let's not put TODOs in the code. I actually do have a personal task in task warrior to remind me to document our schemas. We could also file an issue about it if you like.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

**Comment by abompard** [comment]
Sorry about the unrelated style changes, I had my editor configured to run `Black` automatically on file save and forgot to turn it off. I'll remove those changes.

File: docs/developer/releases.rst (hunk 10 above)
Comment (abompard) [reply docs/developer/releases.rst]: Yeah sorry about that it wasn't intentional.

**Comment by abompard** [comment]
I've removed the stying autofixes and rebased, it should be ready for re-review now.

File: bodhi/tests/server/test_notifications.py (hunk 9 above)
Comment (abompard) [reply bodhi/tests/server/test_notifications.py]: You're correct, we don't have any of those. I'll remove those branches.

**Comment by abompard** [comment]
The `pip-build` job went overtime for some reason.

File: devel/ansible/roles/bodhi/files/.bashrc [hunk 12]
```diff
@@ -16,7 +16,7 @@ alias brestart="sudo systemctl restart bodhi && echo 'The Application is running
 alias bstart="sudo systemctl start bodhi && echo 'The Application is running on http://localhost:6543'"
 alias bstop="sudo systemctl stop bodhi"
 alias bteststyle="flake8-3 && pydocstyle bodhi"
-alias bfedmsg="sudo journalctl -u fedmsg-tail"
+alias bfedmsg="sudo journalctl -u print-messages"
```
Comment (cverna) [inline devel/ansible/roles/bodhi/files/.bashrc:19]: not a big deal but do we want to rename this alias ?

File: devel/ansible/roles/bodhi/files/motd [hunk 13]
```diff
@@ -3,7 +3,7 @@ Welcome to the Bodhi development environment! Here are some helpful commands:
 
 bci:         Run the Bodhi CI test suite.
 bdocs:       Build Bodhi's documentation.
-bfedmsg:     Display the log of Bodhi's fedmsgs.
+bfedmsg:     Display the log of Bodhi's messages on the bus.
```
Comment (cverna) [inline devel/ansible/roles/bodhi/files/motd:6]: If we rename the alias this need to be updated

**Review by cverna — APPROVED** [review]
Just one question, but otherwise :+1: .

Feel free to merge without another review

File: devel/ansible/roles/bodhi/files/.bashrc (hunk 12 above)
Comment (abompard) [reply devel/ansible/roles/bodhi/files/.bashrc:19]: Nah I think it's fine.

**Comment by bowlofeggs** [comment]
This patch is planned for inclusion in the upcoming 4.0.0 release: #3221

**Comment by bowlofeggs** [comment]
Bodhi 4.0.0 beta is built and deployed to staging:

https://copr.fedorainfracloud.org/coprs/bowlofeggs/bodhi-pre-release
https://bodhi.stg.fedoraproject.org/
